src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.cs

```diff
@@ -28,6 +25,10 @@ public HostAvailabilityCheckMiddleware(RequestDelegate next, ILoggerFactory logg
             {
                 Logger.InitiatingHostAvailabilityCheck(_logger);
 
+                // need to ensure the host startup has been triggered
+                var tIgnore = manager.EnsureHostStarted(CancellationToken.None);
```

Comment:
This fixes https://github.com/Azure/azure-functions-host/issues/2773 which my new e2e test was sporadically failing on

Reply:
Why do we need these calls here, though? This is done by the `IHostedService` implementation, which will handle the service/host lifetime.

Was there a scenario where we got to this call without a hosted service initialization?

Reply:
IHostedService is only called on startup. Specialization happens AFTER startup. My end to end test shows this. Also @alrod hit this in #2773 as well